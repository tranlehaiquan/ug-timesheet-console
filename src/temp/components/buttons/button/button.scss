.sked-button {
  @apply sk-leading-normal sk-text-sm sk-font-medium sk-border sk-border-solid sk-px-4 sk-py-2 sk-relative sk-rounded-medium;
  line-height: 1.3;

  &:focus {
    @apply sk-outline-none;
  }

  &--primary {
    @apply sk-bg-blue sk-text-white sk-border-blue sk-tracking-wide;

    &:hover {
      @apply sk-border-blue-dark sk-bg-blue-dark;
    }

    &[disabled] {
      @apply sk-bg-grey-lightest sk-border-grey-lightest;
    }
  }

  &--secondary {
    @apply sk-text-navy-light sk-bg-transparent sk-border-grey-light;
    
    &:hover {
      @apply sk-bg-grey-lightest;
    }

    &[disabled] {
      @apply sk-bg-grey-lightest sk-border-grey-lightest;
    }
  }

  &--transparent {
    @apply sk-text-navy-light sk-bg-transparent sk-border-transparent;

    &:hover {
      @apply sk-text-navy;
    }
  }

  &[disabled] {
    @apply sk-pointer-events-none sk-text-navy-lightest sk-tracking-normal;
  }

  &--compact {
    @apply sk-px-3 sk-py-1;
    line-height: 1.6;
  }
}