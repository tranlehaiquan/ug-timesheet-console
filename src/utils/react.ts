import * as React from 'react'

type IdentityFunction = <T>(fn: T) => T

export const typedMemo: IdentityFunction = React.memo

export const getDisplayName = <T extends any = any>(WrappedComponent: React.ComponentType<T>): string => {
  return WrappedComponent.displayName || WrappedComponent.name || 'Component'
}
