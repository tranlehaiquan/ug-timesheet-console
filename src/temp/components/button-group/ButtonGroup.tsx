import * as React from "react";
import classNames from "classnames";

interface IButtonGroup extends React.HTMLProps<HTMLDivElement> {
  compact?: boolean;
}

export const ButtonGroup = (props: IButtonGroup) => {
  const { compact, className, ...otherProps } = props;
  const classes = classNames(
    "sked-button-group",
    { "sked-button-group--compact": compact },
    className
  );

  return (
    <div className={classes} {...otherProps}>
      {props.children}
    </div>
  );
};
