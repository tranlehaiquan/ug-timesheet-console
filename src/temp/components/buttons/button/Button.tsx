import * as React from 'react'
import classNames from 'classnames'

import { LoadingSpinner } from '../../loader/spinner/LoadingSpinner'

export type ButtonTypes = 'primary' | 'secondary' | 'transparent'

type IButtonProps = React.ButtonHTMLAttributes<HTMLButtonElement> & React.RefAttributes<HTMLButtonElement> & {
  /**
  * The type of button to be displayed. Primary | Secondary | Transparent
  */
  buttonType: ButtonTypes,
  /**
  * Pass additional class names
  */
  className?: string,
  /**
  * Disable the button
  */
  disabled?: boolean
  /**
  * Compact sized buttonâ€”used when space is tight - allowing more room for content.
  */
  compact?: boolean
  /**
  * Loading button state
  */
  loading?: boolean
}

/**
* Button component
*/
export const Button: React.RefForwardingComponent<HTMLButtonElement, IButtonProps> = React.forwardRef<HTMLButtonElement, IButtonProps>(({ buttonType, compact, loading, children, className, ...otherProps }, ref) => {

  const buttonClassMap: { [key in ButtonTypes]: string } = {
    primary: 'sked-button--primary',
    secondary: 'sked-button--secondary',
    transparent: 'sked-button--transparent'
  }

  const classes = classNames(
    'sked-button',
    buttonClassMap[buttonType],
    {
      'sked-button--compact': compact,
      'sk-pointer-events-none': loading
    },
    className
  )

  return (
    <button { ...otherProps }  className={ classes } ref={ ref }>
      <div className={ loading ? 'sk-invisible' : 'sk-visible' }>
        { children }
      </div>
      <div className="sk-pin sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center">
        <LoadingSpinner className={ loading ? 'sk-visible' : 'sk-invisible' } size={ 18 } />
      </div>
    </button>
  )
})
