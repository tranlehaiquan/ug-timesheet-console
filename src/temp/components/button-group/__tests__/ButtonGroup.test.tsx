import * as React from "react";
import { render } from "enzyme";

import { Button } from "../../buttons/button/Button";
import { ButtonGroup } from "../ButtonGroup";

describe("ButtonGroup", () => {
  test("it renders correctly", () => {
    const component = render(
      <ButtonGroup>
        <Button buttonType="transparent" />
        <Button buttonType="primary" />
      </ButtonGroup>
    );
    expect(component).toMatchSnapshot();
  });

  test("it renders compact correctly", () => {
    const component = render(
      <ButtonGroup compact>
        <Button buttonType="transparent" />
        <Button buttonType="primary" />
      </ButtonGroup>
    );
    expect(component).toMatchSnapshot();
  });
});
