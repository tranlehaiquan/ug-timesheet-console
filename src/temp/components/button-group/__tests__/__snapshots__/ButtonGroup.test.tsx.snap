// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup it renders compact correctly 1`] = `
<div
  class="sked-button-group sked-button-group--compact"
>
  <button
    class="sked-button sked-button--transparent"
  >
    <div
      class="sk-visible"
    />
    <div
      class="sk-pin sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center"
    >
      <div
        class="loading-spinner sk-invisible"
      >
        <svg
          class="loading-spinner--circular"
          height="18"
          viewBox="25 25 50 50"
          width="18"
        >
          <circle
            class="loading-spinner--path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="5"
          />
        </svg>
      </div>
    </div>
  </button>
  <button
    class="sked-button sked-button--primary"
  >
    <div
      class="sk-visible"
    />
    <div
      class="sk-pin sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center"
    >
      <div
        class="loading-spinner sk-invisible"
      >
        <svg
          class="loading-spinner--circular"
          height="18"
          viewBox="25 25 50 50"
          width="18"
        >
          <circle
            class="loading-spinner--path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="5"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;

exports[`ButtonGroup it renders correctly 1`] = `
<div
  class="sked-button-group"
>
  <button
    class="sked-button sked-button--transparent"
  >
    <div
      class="sk-visible"
    />
    <div
      class="sk-pin sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center"
    >
      <div
        class="loading-spinner sk-invisible"
      >
        <svg
          class="loading-spinner--circular"
          height="18"
          viewBox="25 25 50 50"
          width="18"
        >
          <circle
            class="loading-spinner--path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="5"
          />
        </svg>
      </div>
    </div>
  </button>
  <button
    class="sked-button sked-button--primary"
  >
    <div
      class="sk-visible"
    />
    <div
      class="sk-pin sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center"
    >
      <div
        class="loading-spinner sk-invisible"
      >
        <svg
          class="loading-spinner--circular"
          height="18"
          viewBox="25 25 50 50"
          width="18"
        >
          <circle
            class="loading-spinner--path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="5"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;
